packages/blockchain-wallet-v4-frontend/src/modals/BuySell/index.tsx

```diff
@@ -70,10 +70,10 @@ class BuySell extends PureComponent<Props, State> {
   }
 
   componentDidMount() {
-    /* eslint-disable */
     this.setState({ show: true })
     this.props.custodialActions.fetchProductEligibilityForUser()
-    /* eslint-enable */
+
+    this.props.recurringBuyActions.fetchPaymentInfo()
```

Comment:
I think it is safe and actually desirable to fetch payment info only once when we enter the buy/sell functionality. Thus, I moved it here, and removed other calls to it within the module.

This place is not perfect since the call will be triggered for sell as well; however, keeping the call in the first buy step (EnterAmount) really messes up the current implementation of reseting period for recurring buys.

Let me know if see any issue with this.

Reply:
You could probably take it a step further and only fetch the payment info when the user clicks the "One Time" button and opens the recurring buy step.

Reply:
Unfortunately, we need this data to show that "One Time" button.

Ideally, I would like payment aggregate to expose this information to us instead of fetching it separate.